File: tests/fixtures/header_injection/python/safe_subscript_set.py
# Safe: request arg routed through `strip_crlf` (a registered
# HEADER_INJECTION sanitizer) before the subscript-set, so
# taint-header-injection stays clean.
from flask import request, make_response


def strip_crlf(raw):
    return raw.replace("\r", "").replace("\n", "")


def handler():
    lang = request.args.get("lang")
    response = make_response("ok")
    response.headers["X-Forwarded-By"] = strip_crlf(lang)
    return response

File: tests/fixtures/header_injection/ruby/safe_subscript_set.rb
# Safe: tainted request value routed through `strip_crlf` (a registered
# HEADER_INJECTION sanitizer) before the subscript-set, so taint-header-injection
# stays clean.
def handle(params, response)
  lang = params["lang"]
  response.headers["X-Forwarded-By"] = strip_crlf(lang)
end
